pkg/kncloudevents/message_sender.go

```diff
@@ -46,41 +39,15 @@ var noRetries = RetryConfig{
 	},
 }
 
-// ConnectionArgs allow to configure connection parameters to the underlying
-// HTTP Client transport.
-type ConnectionArgs struct {
-	// MaxIdleConns refers to the max idle connections, as in net/http/transport.
-	MaxIdleConns int
-	// MaxIdleConnsPerHost refers to the max idle connections per host, as in net/http/transport.
-	MaxIdleConnsPerHost int
-}
-
-func (ca *ConnectionArgs) ConfigureTransport(transport *nethttp.Transport) {
-	if ca == nil {
-		return
-	}
-	transport.MaxIdleConns = ca.MaxIdleConns
-	transport.MaxIdleConnsPerHost = ca.MaxIdleConnsPerHost
-}
-
 type HttpMessageSender struct {
 	Client *nethttp.Client
 	Target string
 }
 
-func NewHttpMessageSender(connectionArgs *ConnectionArgs, target string) (*HttpMessageSender, error) {
-	// Add connection options to the default transport.
-	var base = nethttp.DefaultTransport.(*nethttp.Transport).Clone()
-	connectionArgs.ConfigureTransport(base)
-	// Add output tracing.
-	client := &nethttp.Client{
-		Transport: &ochttp.Transport{
-			Base:        base,
-			Propagation: tracecontextb3.TraceContextEgress,
-		},
-	}
-
-	return &HttpMessageSender{Client: client, Target: target}, nil
+// Deprecated: Don't use this anymore, now it has the same effect of NewHTTPMessageSenderWithTarget
+// If you need to modify the connection args, use ConfigureConnectionArgs sparingly.
+func NewHttpMessageSender(ca *ConnectionArgs, target string) (*HttpMessageSender, error) {
```

Comment:
do we need a backport of NewHTTPMessageSenderWithTarget here or otherwise change the message above?

Reply:
Nah, I don't think it makes sense, the new method is just for clarity for downstream users of this API